nemo/collections/tts/models/magpietts.py

```diff
@@ -97,6 +98,34 @@ def __init__(self, cfg: DictConfig, trainer: 'Trainer' = None):
         if trainer is not None:
             self.world_size = trainer.num_nodes * trainer.num_devices
 
+        # load codec
+        codec_model = AudioCodecModel.restore_from(cfg.get('codecmodel_path'), strict=False)
+        # del codec discriminator to free memory
+        del codec_model.discriminator
+        self._codec_model = codec_model
+        self._codec_model.freeze()  #Lightning does requires_grad = False and self.eval()
+
+        # Reserve special tokens (appended at the end of the codebook after the codec tokens)
+        # (the actual index is this value plus the number of codec tokens - do not use the Enum directy)
+        class SpecialAudioToken(Enum):
+            AUDIO_BOS = 0
+            AUDIO_EOS = 1
+            AUDIO_CONTEXT_BOS = 2
+            AUDIO_CONTEXT_EOS = 3
+            MASK_TOKEN = 4
+            NUM_SPECIAL_TOKENS = 5 # update this if you add more special tokens
```

Comment:
Should we leave this as the source of truth for the number of special tokens? And delete num_reserved_audio_tokens from the configs?

Reply:
Maybe we can add an `override_tokens_per_codebook` param?